godtools/App/Features/ToolDetails/Presentation/ToolDetails/ToolDetailsViewModel.swift

```diff
@@ -0,0 +1,252 @@
+        
+        let mediaType: ToolDetailsMediaType
+        
+        if !resource.attrAboutOverviewVideoYoutube.isEmpty {
+            
+            let playsInFullScreen: Int = 0
+            let playerParameters: [String: Any] = ["playsinline": playsInFullScreen]
+            
+            mediaType = .youtube(videoId: resource.attrAboutOverviewVideoYoutube, playerParameters: playerParameters)
+        }
+        else if !resource.attrAboutBannerAnimation.isEmpty, let filePath = dataDownloader.attachmentsFileCache.getAttachmentFileUrl(attachmentId: resource.attrAboutBannerAnimation)?.path {
+            
+            let resource: AnimatedResource = .filepathJsonFile(filepath: filePath)
+            let viewModel = AnimatedViewModel(animationDataResource: resource, autoPlay: true, loop: true)
+            
+            mediaType = .animation(viewModel: viewModel)
+        }
+        else if let uiImage = dataDownloader.attachmentsFileCache.getAttachmentBanner(attachmentId: resource.attrBannerAbout) {
+            let image: Image = Image(uiImage: uiImage)
+            mediaType = .image(image: image)
+        }
+        else {
+            mediaType = .empty
+        }
+        
+        return mediaType
+    }
+    
+    private func reloadLearnToShareToolButtonState() {
+        
+        guard let primaryLanguage = languageSettingsService.primaryLanguage.value else {
+            return
+        }
+        
+        let determineToolTranslationsToDownload = DetermineToolTranslationsToDownload(
+            resourceId: resource.id,
+            languageIds: [primaryLanguage.id],
+            resourcesCache: dataDownloader.resourcesCache,
+            languagesRepository: languagesRepository
+        )
+        
+        getToolTranslationsUseCase.getToolTranslations(determineToolTranslationsToDownload: determineToolTranslationsToDownload, downloadStarted: { [weak self] in
+            // download started, currently nothing will be shown while this downloads in the background. ~Levi
+        }, downloadFinished: { [weak self] (result: Result<ToolTranslations, GetToolTranslationsError>) in
+            DispatchQueue.main.async { [weak self] in
+                
+                let hidesLearnToShareToolButtonValue: Bool
+                
+                switch result {
+                case .success(let toolTranslations):
+                    
+                    if let primaryLanguageTranslationManifest = toolTranslations.languageTranslationManifests.first {
+                        
+                        hidesLearnToShareToolButtonValue = primaryLanguageTranslationManifest.manifest.tips.isEmpty
+                    }
+                    else {
+                        
+                        hidesLearnToShareToolButtonValue = true
+                    }
+                    
+                case .failure(let error):
```

Comment:
Hey @rachaelblue I'll go ahead and silence that warning since error is not being used.  Might be best to keep warnings like those silenced if not being used for anything.

Reply:
Sounds good!